File: 221125/JavaScript/targetNumber.js
export default function solution(numbers, target) {
  const initial = [numbers[0], numbers[0] * -1];

  const results = numbers.splice(1).reduce((accumulator, number) => {
    const replacing = [];
    accumulator.forEach((interim) => {
      replacing.push(interim + number);
      replacing.push(interim - number);
    });
    return replacing;
  }, initial);

  return results.filter((result) => result === target).length;
}

File: Previous/221125/Java/TargetNumber.java
import java.util.ArrayList;
import java.util.List;

public class TargetNumber {
    public int solution(int[] numbers, int target) {
        List<Integer> accumulated = new ArrayList<>();
        accumulated.add(numbers[0]);
        accumulated.add(numbers[0] * -1);

        for (int i = 1; i < numbers.length; i += 1) {
            int number = numbers[i];
            List<Integer> replacement = new ArrayList<>();
            accumulated.forEach(interim -> {
                replacement.add(interim + number);
                replacement.add(interim - number);
            });
            accumulated = replacement;
        }

        return (int) accumulated.stream()
            .filter(result -> result.equals(target))
            .count();
    }
}
